libraries/core_libs/network/src/tarcap/packets_handlers/dag_sync_packet_handler.cpp

```diff
@@ -58,30 +58,30 @@ void DagSyncPacketHandler::process(const PacketData& packet_data, const std::sha
   }
 
   uint64_t transactions_count = (*it++).toInt<uint64_t>();
-  SharedTransactions new_transactions;
+  std::vector<std::pair<std::shared_ptr<Transaction>, TransactionStatus>> new_transactions;
   std::string transactions_to_log;
   for (uint64_t i = 0; i < transactions_count; i++) {
-    std::shared_ptr<Transaction> trx;
+    std::pair<std::shared_ptr<Transaction>, TransactionStatus> trx;
 
     try {
-      trx = std::make_shared<Transaction>(*it++);
+      trx.first = std::make_shared<Transaction>(*it++);
     } catch (const Transaction::InvalidSignature& e) {
       throw MaliciousPeerException("Unable to parse transaction: " + std::string(e.what()));
     }
 
-    peer->markTransactionAsKnown(trx->getHash());
-    transactions_to_log += trx->getHash().abridged();
-    if (trx_mgr_->isTransactionKnown(trx->getHash())) {
+    peer->markTransactionAsKnown(trx.first->getHash());
+    transactions_to_log += trx.first->getHash().abridged();
+    if (trx_mgr_->isTransactionKnown(trx.first->getHash())) {
       continue;
     }
 
-    if (const auto [is_valid, reason] = trx_mgr_->verifyTransaction(trx); !is_valid) {
+    auto verification = trx_mgr_->verifyTransaction(trx.first);
+    if (verification.first == TransactionStatus::Invalid) {
       std::ostringstream err_msg;
-      err_msg << "DagBlock transaction " << trx->getHash() << " validation failed: " << reason;
-
+      err_msg << "DagBlock transaction " << trx.first->getHash() << " validation failed: " << verification.second;
```

Comment:
hmm, I do not think this is good, maybe I am wrong, but doesn't this mean I can spam the network with transactions with 0 funds, everyone will accept them and no one mark me as malicious? I think we need to do it other way. Maybe we should just save those trx, but not the actual dag block and do nothing

Reply:
I think that we should have some limit on how many transaction with insufficient funds or low nonce we allow from a node and than mark node as malicious if it goes over this threshold. But the whole idea of having insufficient funds and low nonce is to allow dag blocks that might contain them.